src/System.Runtime.Environment/src/OSName.cs

```diff
@@ -0,0 +1,61 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace System.Runtime.InteropServices
+{
+    public struct OSName : IEquatable<OSName>
+    {
+        private readonly string _name;
+
+        private const string WindowsName = "WINDOWS";
+        private const string LinuxName = "LINUX";
+        private const string OSXName = "OSX";
+
+        public static readonly OSName Windows = new OSName(WindowsName);
+        public static readonly OSName Linux = new OSName(LinuxName);
+        public static readonly OSName OSX = new OSName(OSXName);
+
+        public OSName(string name)
+        {
+            if (name == null) throw new ArgumentNullException("name");
+            if (name.Length == 0) throw new ArgumentException(SR.Argument_EmptyValue, "name");
+
+            _name = name;
+        }
+
+        public bool Equals(OSName other)
+        {
+            return string.Equals(other._name, _name, StringComparison.Ordinal);
```

Comment:
OSName is a struct, so there is no concept of null.  It could be defaul(OSName) but the field access would work in this case (you'd get back null).